ClockDisplay.java

```diff
@@ -78,7 +84,17 @@ public String getTime()
      */
     private void updateDisplay()
     {
-        displayString = hours.getDisplayValue() + ":" + 
-                        minutes.getDisplayValue();
+        if (hours.getValue() == 0){
+            displayString = "12:" + minutes.getDisplayValue() + " AM";
+        }
+        else if (hours.getValue() >= 13){
+            hours.setValue(hours.getValue()-12);
+            displayString = hours.getDisplayValue() + ":" + 
+                        minutes.getDisplayValue() + " PM";
+        }
+        else{
+            displayString = hours.getDisplayValue() + ":" + 
+                        minutes.getDisplayValue() + " AM";
+        }    
```

Comment:
This does the work. There are three different statements that need to get updated. Try to only put the pieces that change in the if blocks. use temp variables as needed to hold intermediate values